kotlinpoet/src/commonMain/kotlin/com/squareup/kotlinpoet/ClassName.kt

```diff
@@ -249,28 +246,20 @@ public class ClassName internal constructor(
   message = "Java reflection APIs don't give complete information on Kotlin types. Consider using" +
     " the kotlinpoet-metadata APIs instead."
 )
-@JvmName("get")
-public fun Class<*>.asClassName(): ClassName {
-  require(!isPrimitive) { "primitive types cannot be represented as a ClassName" }
-  require(Void.TYPE != this) { "'void' type cannot be represented as a ClassName" }
-  require(!isArray) { "array types cannot be represented as a ClassName" }
-  val names = mutableListOf<String>()
-  var c = this
-  while (true) {
-    names += c.simpleName
-    val enclosing = c.enclosingClass ?: break
-    c = enclosing
-  }
-  // Avoid unreliable Class.getPackage(). https://github.com/square/javapoet/issues/295
-  val lastDot = c.name.lastIndexOf('.')
-  if (lastDot != -1) names += c.name.substring(0, lastDot)
-  names.reverse()
-  return ClassName(names)
-}
+public expect fun JvmClass<*>.asClassName(): ClassName
+
+/** Returns the class name for `element`. */
+@DelicateKotlinPoetApi(
+  message = "Element APIs don't give complete information on Kotlin types. Consider using" +
+    " the kotlinpoet-metadata APIs instead."
+)
+public expect fun JvmTypeElement.asClassName(): ClassName
```

Comment:
Things that are JVM-specific should not be in common. Moreover, kotlinpoet-metadata is JVM-only due to kotlin-metadata being JVM-only.